RFS/src/main/java/org/opensearch/migrations/bulkload/lucene/LuceneReader.java

```diff
@@ -0,0 +1,229 @@
+package org.opensearch.migrations.bulkload.lucene;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.opensearch.migrations.bulkload.common.RfsLuceneDocument;
+
+import lombok.extern.slf4j.Slf4j;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+@Slf4j
+public class LuceneReader {
```

Comment:
what's the short version of how `LuceneReader` is different from `LuceneIndexReader`?

Reply:
I put all of the concrete code into this file so it would have a consistent logger (@Slf4j only works in classes), whereas the LuceneIndexReader is an interface that Lucene6/7/9 would support.  This also made the interface file simpler to look at and mock out.